### PR TITLE
feat(ops): add keras.ops.image.euclidean_dist_transform

### LINKED ISSUE #22038: Add `euclidean_dist_transform`
This methods becomes very crucial in certian cases, i.e. loss calculation etc. 

- [Euclidean distance transform in tensorflow](https://stackoverflow.com/questions/47140842/euclidean-distance-transform-in-tensorflow)
- [tf-addons adaption](https://github.com/tensorflow/addons/blob/d208d752e98c310280938efa939117bf635a60a8/tensorflow_addons/image/distance_transform.py#L29)
- `from scipy.ndimage import distance_transform_edt`
- https://github.com/tensorflow/tensorflow/issues/24410

### PR DESCRIPTION
**Pull request by rstar327**
## Description

Adds `keras.ops.image.euclidean_dist_transform` — for each non-background pixel of a binary image, computes the Euclidean distance to the nearest background (zero-valued) pixel. Each channel is processed independently. Closes #22038.

This is commonly needed for distance-based loss functions on segmentation masks (boundary loss, signed distance regression, weighted CE) and for distance-aware visualization.

**Signature**

```python
keras.ops.image.euclidean_dist_transform(
    images,           # 3D (H, W, C) or 4D (N, H, W, C) integer tensor
    dtype="float32",  # output float dtype
    data_format=None, # "channels_last" | "channels_first" | None -> config
)
```

**Behavior**

- Returns a tensor of the same shape as `images` with the requested float dtype.
- Per-channel: distance from each non-zero pixel to the nearest zero pixel; zero pixels have distance 0.
- Implementation reuses `scipy.ndimage.distance_transform_edt` (already a Keras dependency) through each backend's appropriate callback mechanism:
  - tensorflow: `tf.numpy_function`
  - jax: `jax.pure_callback`
  - torch / numpy: direct numpy call on device-host buffer
  - openvino: raises `NotImplementedError` (matches the `sobel_edges` precedent)

**API additions**

- `keras.ops.image.euclidean_dist_transform`

**Tests**

`EuclideanDistTransformTest` in `keras/src/ops/image_test.py` covers:

- 3D and 4D inputs, `channels_first` and `channels_last`
- Numeric agreement with the scipy reference on a randomized input
- Closed-form check on a single-background-pixel image (distance equals geometric distance to that pixel)
- Dtype handling (`float32` default, `float16` explicit)
- Input validation (rejects float input, rejects rank-2 input)
- Symbolic call shape/dtype propagation

All 9 tests pass on tensorflow, jax, torch, numpy backends locally. openvino is covered in CI (raises with a clear message and the test class skips).

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by keerthanakadiri** [comment]
Hi @rstar327 ,Can you kindly resolve the conflicts? Thank you!

**Comment by rstar327** [comment]
Resolved the merge conflict

**Comment by hertschuh** [comment]
@rstar327 

I'll be honest with you, I don't see value in this. It's just a wrapper for `scipy.ndimage.distance_transform_edt` on all backends. People can easily call `scipy.ndimage.distance_transform_edt` themselves if they need to.

This op gives the illusion that:
1. it runs on backend native tensors
2. it runs on GPU/TPU
3. it is compilable

But it is none of these things. All ops should meet criteria 1. and 2. and almost all ops should meet criteria 3.
